codalab/rest/bundle_actions.py

```diff
@@ -0,0 +1,49 @@
+from bottle import local, post, request
+from marshmallow import validate
+from marshmallow_jsonapi import Schema, fields
+
+from codalab.common import State, UsageError, precondition
+from codalab.lib.bundle_action import BundleAction
+from codalab.lib.spec_util import validate_child_path, validate_uuid
+from codalab.objects.permission import check_bundles_have_all_permission
+from codalab.server.authenticated_plugin import AuthenticatedPlugin
+
+
+class BundleActionSchema(Schema):
+    id = fields.Integer(dump_only=True, default=None)
+    uuid = fields.String(validate=validate_uuid)
+    type = fields.String(validate=validate.OneOf({BundleAction.KILL, BundleAction.WRITE}))
+    subpath = fields.String(validate=validate_child_path)
+    string = fields.String()
+
+    class Meta:
+        type_ = 'bundle-actions'
+
+
+@post('/bundle-actions', apply=AuthenticatedPlugin())
+def create_bundle_actions():
+    """
+    Sends the message to the worker to do the bundle action, and adds the
+    action string to the bundle metadata.
+    """
+    actions = BundleActionSchema(
+        strict=True, many=True,
+    ).load(request.json).data
+
+    check_bundles_have_all_permission(local.model, request.user, [a['uuid'] for a in actions])
+
+    for action in actions:
+        bundle = local.model.get_bundle(action['uuid'])
+        if bundle.state != State.RUNNING:
```

Comment:
I agree, but in our current worker system, a worker has to be online and assigned to run the bundle before we can send a message. In this function, the following call to `get_bundle_worker` will fail.

Queuing up messages will require some significant redesign -- I think that's not high priority right now.